packages/taco/test/conditions/compound-condition.test.ts

```diff
@@ -205,4 +216,67 @@ describe('validation', () => {
     });
     expect(condition.value.conditionType).toEqual(CompoundConditionType);
   });
+
+  it('rejects invalid operator', () => {
+    const badObj = {
+      operator: 'invalid-operator',
+      operands: [testContractConditionObj, testTimeConditionObj],
+    };
+    expect(() => new CompoundCondition(badObj)).toThrow();
+  });
+
+  it.each(['or', 'and'])(
+    'rejects empty operands for "%s" operator',
+    (operator) => {
+      const badObj = {
+        operator,
+        operands: [],
+      };
+      expect(() => new CompoundCondition(badObj)).toThrow();
+    },
+  );
+
+  it.each(['or', 'and'])(
```

Comment:
Should `not` be in here as well - it should reject non-array operands?
```suggestion
  it.each(['or', 'and', 'not'])(
```